tests/test_vqe.py

```diff
@@ -200,22 +200,30 @@ def amp_embed_and_strong_ent_layer(params, wires=None):
 @pytest.fixture(scope="function", name="mock_device")
 def mock_device_fixture(monkeypatch):
     with monkeypatch.context() as m:
-        m.setattr(qml.Device, "__abstractmethods__", frozenset())
+        m.setattr(qml.devices.LegacyDevice, "__abstractmethods__", frozenset())
         m.setattr(
-            qml.Device, "_capabilities", {"supports_tensor_observables": True, "model": "qubit"}
+            qml.devices.LegacyDevice,
+            "_capabilities",
+            {"supports_tensor_observables": True, "model": "qubit"},
         )
-        m.setattr(qml.Device, "operations", ["RX", "RY", "Rot", "CNOT", "Hadamard", "StatePrep"])
         m.setattr(
-            qml.Device, "observables", ["PauliX", "PauliY", "PauliZ", "Hadamard", "Hermitian"]
+            qml.devices.LegacyDevice,
+            "operations",
+            ["RX", "RY", "Rot", "CNOT", "Hadamard", "StatePrep"],
         )
-        m.setattr(qml.Device, "short_name", "MockDevice")
-        m.setattr(qml.Device, "expval", lambda self, x, y, z: 1)
-        m.setattr(qml.Device, "var", lambda self, x, y, z: 2)
-        m.setattr(qml.Device, "sample", lambda self, x, y, z: 3)
-        m.setattr(qml.Device, "apply", lambda self, x, y, z: None)
+        m.setattr(
+            qml.devices.LegacyDevice,
+            "observables",
+            ["PauliX", "PauliY", "PauliZ", "Hadamard", "Hermitian"],
+        )
+        m.setattr(qml.devices.LegacyDevice, "short_name", "MockDevice")
+        m.setattr(qml.devices.LegacyDevice, "expval", lambda self, x, y, z: 1)
+        m.setattr(qml.devices.LegacyDevice, "var", lambda self, x, y, z: 2)
+        m.setattr(qml.devices.LegacyDevice, "sample", lambda self, x, y, z: 3)
+        m.setattr(qml.devices.LegacyDevice, "apply", lambda self, x, y, z: None)
 
         def get_device(wires=1):
-            return qml.Device(wires=wires)  # pylint:disable=abstract-class-instantiated
+            return qml.LegacyDevice(wires=wires)  # pylint:disable=abstract-class-instantiated
```

Comment:
```suggestion
            return qml.devices.LegacyDevice(wires=wires)  # pylint:disable=abstract-class-instantiated
```

Reply:
I'm curious why no deprecation warnings are raised from here.

Reply:
Looks like it isn't getting used by anything... I'm just deleting the fixture.